integration_tests/suite/test_db_tenant.py

```diff
@@ -299,24 +353,39 @@ def test_delete_access(self, tenant_uuid, policy_uuid):
         result = self.session.query(models.PolicyAccess).get(access_id)
         assert_that(result, equal_to(None))
 
-    def _assert_tenant_matches(self, uuid, name, parent_uuid=ANY_UUID, slug=ANY):
+    def _assert_tenant_matches(
+        self, uuid, name, parent_uuid=ANY_UUID, slug=ANY, domain_names=[]
+    ):
+
         assert_that(uuid, equal_to(ANY_UUID))
         s = self._tenant_dao.session
         tenant = (
-            s.query(models.Tenant.name, models.Tenant.parent_uuid, models.Tenant.slug)
+            s.query(
+                models.Tenant.name,
+                models.Tenant.parent_uuid,
+                models.Tenant.slug,
+            )
             .filter(models.Tenant.uuid == uuid)
             .first()
         )
 
         assert_that(
-            tenant, has_properties(name=name, parent_uuid=parent_uuid, slug=slug)
+            tenant,
+            has_properties(name=name, parent_uuid=parent_uuid, slug=slug),
         )
 
+        if domain_names and len(domain_names) > 0:
+
+            filter_ = models.DomainName.name.in_(domain_names)
+            count_ = s.query(models.DomainName).filter(filter_).count()
+            assert_that(count_, equal_to(len(domain_names)))
```

Comment:
This only check how many domain names there are, not if they match. You should do something similar to this
```suggestion
            names = s.query(models.DomainName.name).filter(filter_).all()
            assert_that(domain_names, contains_inanyorder(names))
```